src/test/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumerWrapper.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.companieshouse.itemhandler.kafka;
+
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.aspectj.lang.annotation.After;
+import org.aspectj.lang.annotation.AfterThrowing;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.core.ProducerFactory;
+import org.springframework.kafka.test.utils.KafkaTestUtils;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+
+@Aspect
+@Service
+public class OrdersKafkaConsumerWrapper {
+    private static final Logger LOGGER = LoggerFactory.getLogger(OrdersKafkaConsumerWrapper.class.getName());
+    private CountDownLatch latch = new CountDownLatch(1);
+    private String orderUri;
+    private String testType = "DEFAULT";
+    @Value("${kafka.broker.addresses}")
+    private String brokerAddresses;
+    private KafkaTemplate<String, String> template;
+    public static final String ORDER_RECEIVED_RETRY_TOPIC = "order-received-retry";
+    private static final String ORDER_RECEIVED_URI = "/order/ORDER-12345";
+
+    @Before(value = "execution(* uk.gov.companieshouse.itemhandler.kafka.OrdersKafkaConsumer.processOrderReceived(..)) && args(message)")
+    public void beforeOrderProcessed(final String message) throws Exception {
+        LOGGER.info("OrdersKafkaConsumer.processOrderReceived() @Before triggered");
+        if (this.testType.equals("RETRY")) {
+            throw new Exception("Mock main listener exception");
```

Comment:
I think you may need to explain (in the code) how/what this does?

Reply:
I have added comments to the aspect implementations.